src/components/OpenWith/OpenWith.js

```diff
@@ -192,9 +200,64 @@ class OpenWith extends PureComponent<Props, State> {
      * @private
      * @return {void}
      */
-    onIntegrationClick(): void {
-        /* no-op */
-    }
+    onIntegrationClick = (appIntegrationId: string): void => {
+        const { fileId }: Props = this.props;
+        this.api
+            .getAppIntegrationsAPI(false)
+            .execute(
+                appIntegrationId,
+                fileId,
+                this.executeIntegrationSuccessHandler,
+                this.executeIntegrationErrorHandler,
+            );
+    };
+
+    /**
+     * Opens the integration in a new tab based on the API data
+     *
+     * @private
+     * @param {ExecuteAPI} executeData - API response on how to open an executed integration
+     * @return {void}
+     */
+    executeIntegrationSuccessHandler = (executeData: ExecuteAPI) => {
+        switch (executeData.method.toLowerCase()) {
```

Comment:
The `toLowerCase()` here tripped me up for a second when I got to the case statements below. Should we keep everything uppercase?

Reply:
yes, we should not lower case it. I was trying to avoid making an extra constant.